File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
input_arr = []

result = []
while True:
  n = input()
  if n == '0':
    break
  input_arr.append(list(n))

for arr in input_arr:
  ans = True
  for i in range(len(arr) // 2):
    if arr[i] == arr[len(arr) - i - 1]:
      continue
    else:
      ans = False
  result.append(ans)

for ans in result:
  if ans:
    print('yes')
  else:
    print('no')

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.cc
#include <iostream>
#include <string>

using namespace std;

string str;
int n;

int main(void) {
	cin.tie(0)->sync_with_stdio(false);

	while (1) {
		cin >> str;

		if (str == "0") {
			break;
		}

		bool result = true;
		for (int i = 0; i < str.size() / 2; i++) {
			if (str[i] != str[str.size() - i - 1]) {
				result = false;
				break;
			}
		}
		cout << (result ? "yes" : "no") << "\n";
	}

	return 0;
}